### PR TITLE
Surface kube config in scheduler framework handle

### LINKED ISSUE #96427: API to access kube-scheduler's options and configuration parameters
**What would you like to be added**:

Provide APIs  for out-of-tree scheduler plugins to access kube-scheduler's options and configuration parameters, such as `kubeconfig` path and `Master`. 

**Why is this needed**:

Out-of-tree scheduler plugins often need `kubeconfig` path and `Master` information to create `clientset` and `informers`. A plugin has to process and manage these parameters separately, e.g., using additional plugin arguments. It puts additional burden on developers. Users have to provide addiontal parameters as well.

As these parameters are already available as the command line arguments and/or in the scheduler configuration, it will simplify the developer's work and improve user experience if a plugin can have a simple way to access `kubeconfig` path and `Master` information from the existing configurations/options data.

### PR DESCRIPTION
**Pull request by Huang-Wei**
#### What type of PR is this?

/kind feature
/sig scheduling

#### What this PR does / why we need it:

Surface kube config in scheduler framework handle, so that developers can leverage this to build CRD informers easily.

#### Which issue(s) this PR fixes:

Fixes #96427

#### Does this PR introduce a user-facing change?

```release-note
Kube config is now exposed in the scheduler framework handle. Out-of-tree plugins can leverage that to build CRD informers easily.
```

### CODE REVIEW

File: pkg/scheduler/apis/config/testing/compatibility_test.go [hunk 1]
```diff
@@ -1379,6 +1379,7 @@ func TestCompatibility_v1_Scheduler(t *testing.T) {
 
 			sched, err := scheduler.New(
 				client,
+				nil,
```
Comment (alculquicondor) [inline pkg/scheduler/apis/config/testing/compatibility_test.go:1382]: can we make it an option instead?

File: test/integration/scheduler/scheduler_test.go [hunk 2]
```diff
@@ -60,8 +60,8 @@ func TestSchedulerCreationFromConfigMap(t *testing.T) {
 
 	ns := framework.CreateTestingNamespace("configmap", s, t)
 	defer framework.DeleteTestingNamespace(ns, s, t)
-
-	clientSet := clientset.NewForConfigOrDie(&restclient.Config{Host: s.URL, ContentConfig: restclient.ContentConfig{GroupVersion: &schema.GroupVersion{Group: "", Version: "v1"}}})
+	kubeConfig := &restclient.Config{Host: s.URL, ContentConfig: restclient.ContentConfig{GroupVersion: &schema.GroupVersion{Group: "", Version: "v1"}}}
+	clientSet := clientset.NewForConfigOrDie(kubeConfig)
```
Comment (alculquicondor) [inline test/integration/scheduler/scheduler_test.go:64]: Are these tests used by the scheduler plugins repo?
I prefer we leave this unchanged if we are not going to use the kubeconfig.

File: test/integration/util/util.go [hunk 3]
```diff
@@ -75,7 +75,7 @@ func StartApiserver() (string, ShutdownFunc) {
 
 // StartScheduler configures and starts a scheduler given a handle to the clientSet interface
 // and event broadcaster. It returns the running scheduler, podInformer and the shutdown function to stop it.
-func StartScheduler(clientSet clientset.Interface, cfg *kubeschedulerconfig.KubeSchedulerConfiguration) (*scheduler.Scheduler, coreinformers.PodInformer, ShutdownFunc) {
+func StartScheduler(clientSet clientset.Interface, kubeConfig *restclient.Config, cfg *kubeschedulerconfig.KubeSchedulerConfiguration) (*scheduler.Scheduler, coreinformers.PodInformer, ShutdownFunc) {
```
Comment (alculquicondor) [inline test/integration/util/util.go:78]: These changes should be fine as external plugins can use them as a library

File: pkg/scheduler/apis/config/testing/compatibility_test.go (hunk 1 above)
Comment (Huang-Wei) [reply pkg/scheduler/apis/config/testing/compatibility_test.go]: sg, so that the logic that doesn't care about this can stay unchanged.

File: test/integration/scheduler/scheduler_test.go (hunk 2 above)
Comment (Huang-Wei) [reply test/integration/scheduler/scheduler_test.go]: Reverted as these tests don't use the kube config.

File: pkg/scheduler/factory_test.go [hunk 4]
```diff
@@ -418,8 +418,7 @@ func TestCreateFromConfig(t *testing.T) {
 					if diff := cmp.Diff(tc.wantPlugins, p.Plugins); diff != "" {
 						t.Errorf("unexpected plugins diff (-want, +got): %s", diff)
 					}
-				}),
-			)
+				}))
```
Comment (alculquicondor) [inline pkg/scheduler/factory_test.go:421]: revert just to avoid the unrelated change

**Review by alculquicondor — COMMENTED** [review]
good for squash

**Comment by Huang-Wei** [comment]
/milestone v1.22

@oomichi would you mind reviewing the changes under `test/integration`? Thanks in advance!
